stepengine-api/src/main/java/org/tweetwallfx/stepengine/api/VisualizationRegistry.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2019 TweetWallFX
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.tweetwallfx.stepengine.api;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.ServiceLoader;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.StreamSupport;
+import org.tweetwallfx.config.Configuration;
+import org.tweetwallfx.stepengine.api.config.StepEngineSettings;
+
+public class VisualizationRegistry {
+
+    public static VisualizationRegistry INSTANCE = new VisualizationRegistry();
+    private final Map<String, StepEngineSettings.VisualizationSetting> visualizationSettingsMap
+            = Configuration.getInstance().getConfigTyped(StepEngineSettings.CONFIG_KEY, StepEngineSettings.class).getVisualizationSettings();
+
+    private final Map<String, Visualization> visualizationCache = new ConcurrentHashMap<>();
+
+    private VisualizationRegistry() {
+    }
+
+    @SuppressWarnings("unchecked")
+    public <V extends Visualization> V getVisualization(String identifier, Class<V> visualizationClass) {
+        Visualization vis = getVisualizationInternal(identifier);
+        if (!visualizationClass.isInstance(vis)) {
+            throw new IllegalArgumentException("Visualization identified by " + identifier + " is not of expected type " + visualizationClass.getName() + " but of type " + vis.getClass().getName());
+        }
+        return (V) vis;
```

Comment:
If you use `visualizationClass.cast(vis)` wouldn't you not need the SuppressWarnings annotation?